src/utils/time-utils.ts

```diff
@@ -41,18 +41,24 @@ export const getHours = (minutes: number) =>
  * @returns formatted timespan in the following formats (DD.MM.YYYY – DD.MM.YYYY), (YYYY/WW), (YYYY/MM)
  */
 export const formatTimePeriod = (timespan: string[] | undefined) => {
-  if (!timespan) return null;
+  if (!timespan) {
+    return null;
+  }
 
-  if (timespan[0].length > 4) {
-    const startDate = DateTime.fromJSDate(new Date(timespan[0])).toLocaleString(
-      DateTime.DATE_SHORT
-    );
-    const endDate = DateTime.fromJSDate(new Date(timespan[1])).toLocaleString(DateTime.DATE_SHORT);
-    return `${startDate} – ${endDate}`; //All time
-  } else if (timespan.length > 2) {
-    return `${timespan[0]}/${timespan[2]}`; //Month
+  switch (timespan.length) {
+    case 1:
+      return `${timespan[0]}`; // Year
+    case 2:
+      return timespan[0].length > 4
+        ? `${DateTime.fromJSDate(new Date(timespan[0])).toLocaleString(
+            DateTime.DATE_SHORT
+          )} – ${DateTime.fromJSDate(new Date(timespan[1])).toLocaleString(DateTime.DATE_SHORT)}` // All time
```

Comment:
Rather than using the methods inline like this for the "allTime" I would suggest defining the start and end date as variables before the return, as was done previously. Improves code readability

Reply:
added the start and end date as variables